src/steps/devices/index.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  createDirectRelationship,
+  Entity,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+} from '@jupiterone/integration-sdk-core';
+
+import { createAPIClient } from '../../client';
+import { IntegrationConfig } from '../../config';
+import {
+  ACCOUNT_ENTITY_KEY,
+  Entities,
+  Relationships,
+  Steps,
+} from '../constants';
+import { createDeviceEntity } from './converter';
+
+export async function fetchDevices({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await apiClient.iterateDevices(async (device) => {
+    const deviceEntity = createDeviceEntity(device);
+
+    if (deviceEntity) {
+      await jobState.addEntity(deviceEntity);
+
+      await jobState.addRelationship(
+        createDirectRelationship({
+          _class: RelationshipClass.HAS,
+          from: (await jobState.getData(ACCOUNT_ENTITY_KEY)) as Entity,
+          to: deviceEntity,
+        }),
+      );
+
+      const userEntity = await jobState.findEntity(
+        `${device.user.name}-${device.user.id.toString()}`,
+      );
+
+      if (userEntity)
+        await jobState.addRelationship(
+          createDirectRelationship({
+            _class: RelationshipClass.HAS,
+            from: userEntity,
+            to: deviceEntity,
+          }),
+        );
+    }
+  });
```

Comment:
2 Suggestions:

1) The _key value thingy (you'll see it in other comments)

2) Can we extract this into a separate step called `buildUsersAndDevicesRelationships`? We should always strive for as many as possible self-contained steps - in case something breaks or some entity is not found, the integration will try to create whatever is possible that way!